dotnet/src/SemanticKernel.Abstractions/Contents/EmbeddingContent.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represent embedding content.
+/// </summary>
+/// <typeparam name="TEmbedding">The numeric type of the embedding data.</typeparam>
+[Experimental("SKEXP0001")]
+public sealed class EmbeddingContent<TEmbedding> : KernelContent where TEmbedding : unmanaged
+{
+    /// <summary>
+    /// Represents embedding content.
+    /// </summary>
+    /// <param name="innerContent">Inner content</param>
+    /// <param name="data">The embedding data</param>
+    /// <param name="modelId">The model ID used to generate the content</param>
+    /// <param name="metadata">Additional metadata</param>
+    public EmbeddingContent(
+        object? innerContent,
```

Comment:
For `Content` classes the pattern is to have `innerContent` parameter as optional with default value - `null`.